src/NServiceBus.Core/Routing/EndpointInstances.cs

```diff
@@ -42,6 +42,14 @@ internal Task<IEnumerable<EndpointInstance>> FindInstances(string endpoint)
                 }
             }
 
+            if (dynamicInstances.Count == 0)
```

Comment:
Since `dynamicInstances` is already created, would it make more sense to just add this instance to the collection  instead of creating a new array here?